src/opart_gaussian.c

```diff
@@ -15,21 +31,97 @@ int opart_gaussian(int n_data, double *data_ptr, double penalty, double *cost_pt
 	  return NUM_OF_DATA_VALUES_LESS_THAN_ZERO;
 	}
 
+	int i;
+	int j;
+	int temp;
+	int maxPos;
+
+	//allocate dynamic memory for global variables
+	sums = (double*)malloc((n_data + 1)*(sizeof(double)));
+	sums_sq = (double*)malloc((n_data + 1)*(sizeof(double)));
+	dp = (double*)malloc((n_data + 1)*(sizeof(double)));
+	positions = (int*)malloc((n_data + 1)*(sizeof(int)));
```

Comment:
please remove malloc/free calls, and instead add arguments so R can perform the dynamic memory allocation for you (thereby decreasing any risk of memory leaks)